app/controllers/application_controller.rb

```diff
@@ -3,19 +3,26 @@ class ApplicationController < ActionController::Base
 
   helper_method :logged_in?, :current_user # Give views access to these application controller methods.
 
+  def require_user
+    if !logged_in?
+      flash[:error] = "You must be logged in to do that."
+      redirect_to front_path
+    end
+  end
+
   # Returns a user object or nil.
   def current_user
     if logged_in?
-      User.find(session[:id])
+      User.find(session[:user_id])
```

Comment:
Doh! I'm sorry, I was thinking this was your method for checking if the user was logged in. My apologies.